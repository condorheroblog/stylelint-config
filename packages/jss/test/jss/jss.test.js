import { readFileSync } from "node:fs";
import { join } from "node:path";
import stylelint from "stylelint";
import { beforeEach, describe, expect, test } from "vitest";

import basicConfig from "../..";

const config = { ...basicConfig, ignoreFiles: [] };
const validJsxPath = join(__dirname, "./Valid.tsx");
const invalidJsxPath = join(__dirname, "./Invalid.tsx");
const validJsx = readFileSync(validJsxPath, "utf-8");
const invalidJsx = readFileSync(invalidJsxPath, "utf-8");

describe("flags no warnings in JSX with valid css", () => {
  let result;

  beforeEach(() => {
    result = stylelint.lint({
      config,
      files: validJsxPath,
      // fix: Error: Could not find "@condorhero/stylelint-config-css". Do you need a `configBasedir`?
      configBasedir: __dirname,
    });
  });

  test("did not error", () => {
    return result.then(data => expect(data.errored).toBeFalsy());
  });

  test("flags no warnings", () => {
    return result.then(data => expect(data.results[0].warnings).toHaveLength(0));
  });
});

describe("flags warnings in HTML with invalid css", () => {
  let result;

  beforeEach(() => {
    result = stylelint.lint({
      config,
      files: invalidJsxPath,
      // fix: Error: Could not find "@condorhero/stylelint-config-css". Do you need a `configBasedir`?
      configBasedir: __dirname,
    });
  });

  test("did error", () => {
    return result.then(data => expect(data.errored).toBeTruthy());
  });

  test("flags one warning", () => {
    return result.then(data => expect(data.results[0].warnings).toHaveLength(3));
  });

  test("correct warning text", () => {
    return result.then(data =>
      expect(data.results[0].warnings[1].text).toBe(
        "Unexpected empty line (no-empty-first-line)",
      ),
    );
  });

  test("correct rule flagged", () => {
    return result.then(data =>
      expect(data.results[0].warnings[1].rule).toBe("no-empty-first-line"),
    );
  });

  test("correct severity flagged", () => {
    return result.then(data => expect(data.results[0].warnings[1].severity).toBe("error"));
  });

  test("correct line number", () => {
    return result.then(data => expect(data.results[0].warnings[1].line).toBe(3));
  });

  test("correct column number", () => {
    return result.then(data => expect(data.results[0].warnings[1].column).toBe(25));
  });
});

describe("jss pieces to be tested", () => {
  test("valid.tsx", () => {
    return expect(validJsx).toMatchSnapshot();
  });

  test("invalid.tsx", () => {
    return expect(invalidJsx).toMatchSnapshot();
  });
});
