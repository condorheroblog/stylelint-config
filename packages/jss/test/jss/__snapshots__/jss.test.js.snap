// Vitest Snapshot v1

exports[`jss pieces to be tested > invalid.tsx 1`] = `
"import styled from \\"@emotion/styled\\";

const Title = styled.h2\`
  margin: 0 0 1rem;
  font-size: 1.5rem;
  width: calc(5% - 10em);
\`;

function Invalid() {
  return (
    <>
      <Title>Learn React</Title>
      <div style={{widht: \\"22px\\",top: 0}} />
    </>
  );
}

export default Invalid;
"
`;

exports[`jss pieces to be tested > valid.tsx 1`] = `
"import styled from \\"@emotion/styled\\";

const Title = styled.h2\`width: calc(5% - 10em);
  margin: 0 0 1rem;
  font-size: 1.5rem;
\`;

function Valid() {
  return (
    <>
      <Title>Learn React</Title>
      <div style={{ width: \\"22px\\", top: 0 }} />
    </>
  );
}

export default Valid;
"
`;
